backend/src/main/java/com/yagubogu/checkin/repository/CustomCheckInRepositoryImpl.java

```diff
@@ -0,0 +1,203 @@
+                .where(filterByTeam(member.team, teamFilter))
+                .groupBy(member.id, member.nickname, member.imageUrl, member.team.shortName)
+                .orderBy(score.desc()).limit(limit).fetch();
+    }
+
+    public VictoryFairyRank findMyRanking(final double m, final double c, final Member targetMember, final int year,
+                                          final TeamFilter teamFilter) {
+        QCheckIn checkIn = QCheckIn.checkIn;
+        QGame game = QGame.game;
+        QMember member = QMember.member;
+
+        // w: 개인의 승리 횟수
+        NumberExpression<Long> wins = calculateWinCounts(checkIn, game);
+        NumberExpression<Long> total = game.id.count().coalesce(0L);
+        NumberExpression<Double> score = calculateWinRankingScore(m, c, wins, total);
+
+        NumberExpression<Double> calculatePercent = wins.sum().multiply(100.0).divide(total).doubleValue();
+
+        NumberExpression<Double> safeWinPercent = getSafeWinPercent(total, calculatePercent);
+
+        return jpaQueryFactory.select(
+                        Projections.constructor(VictoryFairyRank.class, score, member.nickname, member.imageUrl,
+                                member.team.shortName, safeWinPercent))
+                .from(member)
+                .leftJoin(checkIn).on(checkIn.member.eq(member))
+                .leftJoin(game).on(checkIn.game.eq(game), isFinished(game), isBetweenYear(game, year))
+                .where(member.eq(targetMember), filterByTeam(member.team, teamFilter))
+                .groupBy(member.id, member.nickname, member.imageUrl, member.team.shortName)
+                .fetchOne();
+    }
+
+    private NumberExpression<Double> getSafeWinPercent(final NumberExpression<Long> total,
+                                                       final NumberExpression<Double> calculatePercent) {
+        return new CaseBuilder().when(total.gt(0)).then(calculatePercent).otherwise(0.0);
+    }
+
+    public int calculateMyRankingOrder(final double targetScore, final double m, final double c, final int year,
+                                       final TeamFilter teamFilter) {
+        QCheckIn checkIn = QCheckIn.checkIn;
+        QGame game = QGame.game;
+        QMember member = QMember.member;
+        QTeam team = QTeam.team;
+
+        // w: 개인의 승리 횟수
+        NumberExpression<Long> wins = calculateWinCounts(checkIn, game);
+        NumberExpression<Long> total = checkIn.count();
+        NumberExpression<Double> score = calculateWinRankingScore(m, c, wins, total);
+
+        Expression<Double> myScore = Expressions.constant(targetScore);
+        return jpaQueryFactory.selectOne()
+                .from(member)
+                .leftJoin(checkIn).on(checkIn.member.eq(member))
+                .leftJoin(game).on(checkIn.game.eq(game))
+                .where(isFinished(game), isBetweenYear(game, year), filterByTeam(team, teamFilter))
```

Comment:
```suggestion
.leftJoin(checkIn).on(checkIn.member.eq(member))
.leftJoin(game).on(
checkIn.game.eq(game),
isFinished(game),
isBetweenYear(game, year)
)
.leftJoin(checkIn.team, team)
.where(filterByTeam(team, teamFilter))
```

game에 대한 조건이 where절에 포함되어있는데, 이러면 game의 필드가 null인 부분이 무시돼서 결과에 isFinished가 false이거나 isBtweenYear 가 false인 game은 없어질 것으로 예상이 돼요!

Reply:
그렇네요! 수정했습니다 :)